vividus-tests/src/main/resources/story/integration/JWT.story

```diff
@@ -8,10 +8,12 @@ Scenario: Decode/encode JWTs, validate extracked jsons
 Given I initialize scenario variable `JWT` with value `#{loadResource(JWT.txt)}`
 Given I initialize scenario variable `header` with value `#{decodeFromBase64(#{eval(`${JWT}`.replaceFirst("([^.]+).*", "$1"))})}`
 Given I initialize scenario variable `payload` with value `#{decodeFromBase64(#{eval(`${JWT}`.replaceFirst(".*(?<=\.)(.*?)(?=\.).*", "$1"))})}`
-Given I initialize scenario variable `encodedSignature` with value `#{eval(`${JWT}`.replaceFirst(".*(?<=\.)([^.]+)$", "$1"))}`
 When I save JSON element from `${header}` by JSON path `alg` to scenario variable `alg`
 Then `#{removeWrappingDoubleQuotes(${alg})}` is = `HS256`
 Then number of JSON elements from `${header}` by JSON path `typ` is = 1
 Then JSON element from `${payload}` by JSON path `sub` is equal to `"1234567890"`TREATING_NULL_AS_ABSENT
 Then number of JSON elements from `${payload}` by JSON path `name` is = 1
-Then `#{encodeToBase64(${header})}.#{eval(`#{encodeToBase64(${payload})}`.replaceFirst("==", ""))}.${encodedSignature}` is = `${JWT}`
+Given I initialize scenario variable `encodedHeader` with value `#{encodeToBase64(${header})}`
+Given I initialize scenario variable `encodedPayload` with value `#{eval(`#{encodeToBase64(${payload})}`.replaceFirst("==", ""))}`
+When I sign data `${encodedHeader}.${encodedPayload}` using HS256 algorithm with key `secretKey` and save result to scenario variable `encodedSignature`
```

Comment:
Does it make sense to create a single step that will generate a full JWT?

Reply:
updated